### DISCUSSION TITLE
Prevent controller button go out of application

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [axmolengine/axmol#1998] [General]
Hi,

How to prevent controller button go out of application in AppleTV?

When you press circle button from Playstation controller it go out of application. I want capture it to show a pause screen with exit option or continue.

I have a ready project sample:
https://github.com/paulocoutinhox/axmol-controller-test

I tried stopPropagation, but don't works:

```cpp
case Controller::BUTTON_B: {
        event->stopPropagation();

        auto scene = game::factory::SceneFactory::create("main-menu");
        Director::getInstance()->replaceScene(TransitionFade::create(0.5, scene, Color3B(0, 0, 0)));
        break;
}
```

It stop on breakpoint, back to the scene, but go out from game and go to appletv home screen too.

<img width="1012" alt="image" src="https://github.com/axmolengine/axmol/assets/395096/a761b335-7f2d-4947-864d-3efcb38a53a4">

Can anyone help me?

**Comment by paulocoutinhox** [comment]
https://github.com/axmolengine/axmol/assets/395096/0c1a17db-3599-4b1a-b5ce-bfb2d55f197a

**Reply by paulocoutinhox** [reply]
It stop on "shouldStopPropagation = true" but still go out from app:

<img width="1840" alt="image" src="https://github.com/axmolengine/axmol/assets/395096/12942168-0708-41b7-b84c-679a4801d811">

**Reply by rh101** [reply]
> Btw, could it be that Apple TV Os hard reads it meaning it always listens/reacts to that button unless changed from the user settings? Could it me almost like Win+L key behavior on Windows, no other user level privileged app can prevent Windows from reacting to this Win+L key unless changed from the settings.

Precisely this.  @paulocoutinhox You've assumed that this is something controlled within Axmol, but that is unlikely to be the case, because it seems to be used by the OS as a 'Home' button.  There may be a way to intercept the OS command being triggered by this.

**Reply by paulocoutinhox** [reply]
Yes, I know. In debug I see that is button B that we receive. I want know if anyone implement something that solve this problem and intercept it. I want control the flow until go to main menu, where I will let the default behavior.

[a comment by an automated account is not shown]
